flink-table-store-core/src/test/java/org/apache/flink/table/store/table/FileStoreTableTestBase.java

```diff
@@ -296,6 +299,52 @@ public void testWriteWithoutCompaction() throws Exception {
         }
     }
 
+    @Test
+    public void testReadCompactedSnapshot() throws Exception {
+        writeCompactData();
+        FileStoreTable table =
+                createFileStoreTable(conf -> conf.set(CoreOptions.READ_COMPACTED, true));
+
+        DataTableScan.DataFilePlan plan = table.newScan().plan();
+        Snapshot compactedSnapshot = table.snapshotManager().snapshot(plan.snapshotId);
+        Iterator<Snapshot> snapshotIterator = table.snapshotManager().snapshots();
+        while (snapshotIterator.hasNext()) {
+            Snapshot snapshot = snapshotIterator.next();
+            if (snapshot.commitKind() == Snapshot.CommitKind.COMPACT) {
+                assertThat(snapshot.id()).isLessThanOrEqualTo(compactedSnapshot.id());
+            }
+        }
+
+        assertThat(compactedSnapshot.commitKind()).isEqualTo(Snapshot.CommitKind.COMPACT);
+        List<Split> splits = plan.splits();
+        TableRead read = table.newRead();
+        assertThat(getResult(read, splits, binaryRow(1), 0, BATCH_ROW_TO_STRING).size())
+                .isGreaterThan(0);
+        assertThat(getResult(read, splits, binaryRow(2), 0, BATCH_ROW_TO_STRING).size())
+                .isGreaterThan(0);
```

Comment:
The latest snapshot is a compacted snapshot, how do you make sure your implementation works (instead of just reading the latest snapshot)? Change this test.